cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/realtime/source/JmsSource.java

```diff
@@ -198,7 +198,10 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     String pluginId = getPluginId();
     Class<Object> driver  = pipelineConfigurer.usePluginClass(config.jmsPluginType, config.jmsPluginName, pluginId,
                                                               PluginProperties.builder().build());
-    Preconditions.checkArgument(driver != null, "JMS Initial Connection Factory Context class must be found.");
+    Preconditions.checkArgument(driver != null,
+                                "Unable to find JMS Initial Connection Factory Context class. Please make sure that " +
+                                  "the plugin '%s' of type '%s' containing the driver has been installed correctly.",
+                                config.jmsPluginType, config.jmsPluginName);
```

Comment:
Based on your message: "the plugin '%s' of type '%s'", isn't this backwards? Shouldn't it be:
```
config.jmsPluginName, config.jmsPluginType);